pkg/antctl/command_definition.go

```diff
@@ -400,6 +400,9 @@ func (cd *commandDefinition) decode(r io.Reader, single bool) (interface{}, erro
 	if err != nil {
 		return nil, err
 	}
+	if reflect.Indirect(ref).IsZero() {
+		return "", nil
+	}
```

Comment:
This is too specific to your use case IMO. A "zero" struct may not always mean "no object".
Honestly for this command, it would maybe make more sense to return a 404 Not Found in the API if there is no effective BGP policy.

Reply:
I think returning a 404 Not Found is not consistent as there are some other commands like `antctl get addressgroup/appliedtogroup/netpol` returning empty string as output without 404 when there is no resource. I think it is expected that a command should return zero exit code and run successfully when there is no resource. There is a e2e test for the same expectation. https://github.com/antrea-io/antrea/blob/b96d22f213ab58869425daf66d3a9b55ea6ba6ba/test/e2e/antctl_test.go#L112.

Reply:
I think your initial views were also similar https://github.com/antrea-io/antrea/pull/6646#discussion_r1752506508

Reply:
I think the comparison is not great
IMO, this command is more similar to `antctl get addressgroup <name>` than it is to `antctl get addressgroup`, because we are querying a specific object (the "effective" bgppolicy) and not a list of objects. And `antctl get addressgroup <name>` will return an error if there is no matching resource.

My original comment was more about `antctl` output formatting than it was about the API response.
We could in theory have the API return a 404, and the output be an empty table. But IMO, an error message like "there is no effective policy applied to the Node" with a non-zero error code is more appropriate. It is also easier to consume programmatically.

In any case, a 404 for the API response seems more appropriate than a "zero" struct.

Reply:
done, thanks @antoninbas